tests/test_linux.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+
+SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+
+# Variables and environment variables
+LINUX_MAJOR_VERSION=$(echo "${VERIFICATION_TEST_LINUX_VERSION}" | sed -E 's/\..*//g')
+LINUX_BASE="linux-${VERIFICATION_TEST_LINUX_VERSION}"
+LINUX_TAR="${LINUX_BASE}.tar"
+LINUX_TAR_XZ="${LINUX_TAR}.xz"
+LINUX_TAR_SIGN="${LINUX_TAR}.sign"
+
+mkdir -p kernelbuild
+cd kernelbuild
+
+# Download tarball
+if [ -d "${SCRIPT_DIR}/kernelbuild" ]; then
+	# This is for debugging on local system, to not download over 100MB over and over again
+	cp -r "${SCRIPT_DIR}/kernelbuild/"* ./
+else
+	wget --quiet "https://cdn.kernel.org/pub/linux/kernel/v${LINUX_MAJOR_VERSION}.x/${LINUX_TAR_XZ}"
+	wget --quiet "https://cdn.kernel.org/pub/linux/kernel/v${LINUX_MAJOR_VERSION}.x/${LINUX_TAR_SIGN}"
+fi
+
+# un-xz
+if [ ! -f "${LINUX_TAR}" ]; then
+	# LINUX_TAR_XZ -> LINUX_TAR
+	unxz --keep "${LINUX_TAR_XZ}" > /dev/null
+fi
+
+# Verify
+gpg2 --locate-keys torvalds@kernel.org gregkh@kernel.org
+gpg2 --verify "${LINUX_TAR_SIGN}"
+
+# un-tar
+if [ ! -d "${LINUX_BASE}" ]; then
+	# LINUX_TAR -> LINUX_BASE
+	tar -xvf "${LINUX_TAR}"
+fi
+
+# Make
+cd "${LINUX_BASE}"
+cp "${SCRIPT_DIR}/linux_${VERIFICATION_TEST_LINUX_VERSION}/linux.defconfig" ./.config
+make olddefconfig
```

Comment:
According to `make help`, `olddefconfig` = `Same as oldconfig but sets new symbols to their default value without prompting`

Reply:
Ah, then I remembered it incorrectly! Though you would still need to update the dotconfig more often and there are also several lines that are not relevant. And if it also changes defaults in the dotconfig you would create unnecessary diffs. While the olddefconfig workflow is good for local installations and setups of the Linux Kernel I would rather prefer the `make savedefconfig` and then `make xyz_defconfig` workflow.

Reply:
FYI, I am not disputing `config` vs `defconfig`. I am on board with switch to `defconfig`.

Reply:
I just wanted to underline that the olddefconfig flow is totally fine in personal setups but makes little sense in CI as it causes overhead, so all good :)

Reply:
I am at a loss. I got `./defconfig` file by running `make savedefconfig`. How do I compile kernel with it?

Reply:
`cp ./defconfig ./.config; make`?

Reply:
> `cp ./defconfig ./.config; make`?

Actually this would require `make olddefconfig` (`cp ./defconfig ./.config; make olddefconfig; make`) because otherwise you get prompted.

Reply:
There are two ways:
1. Way: Run `KBUILD_DEFCONFIG=/path/to/your/defconfig make defconfig`
2. Way: Copy the defconfig file to `arch/$ARCH/configs/custom_defconfig` and then run `make custom_defconfig` ("custom" can be any string)

After that run `make` as usual for building as both ways create the `.config` file for you.

Reply:
> > `cp ./defconfig ./.config; make`?
> 
> Actually this would require `make olddefconfig` (`cp ./defconfig ./.config; make olddefconfig; make`) because otherwise you get prompted.

Yes, because copying a defconfig to a dotconfig does not work, as they are two different things. The dotconfig should always be generated by the buildsystem and manually created for CI purposes (e.g. by copying).

Reply:
> > There are two ways:
> > ```
> > 1. Way: Run `KBUILD_DEFCONFIG=/path/to/your/defconfig make defconfig`
> > 
> > 2. Way: Copy the defconfig file to `arch/$ARCH/configs/custom_defconfig` and then run `make custom_defconfig` ("custom" can be any string)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > After that run `make` as usual for building as both ways create the `.config` file for you.
> 
> Awersome! Thanks.
> 
> I was looking around for any documentation to `KBUILD_DEFCONFIG` (actually `KBUILD_CONFIG` based on our offline discussion), but found none.

Well there is an inline comment about it in the main Makefile, if you can call that documentation :grimacing:

Reply:
Yeah, I did not dig that deep, it is over 2k lines long. I was surprised (and happy) to find `make help` is a thing, but still rather confusing.

Reply:
`KBUILD_DEFCONFIG=./linux.defconfig make defconfig` is not providing expected results. It just defaults to `x86_64_defconfig`.
```
*** Default configuration is based on 'x86_64_defconfig'
#
# configuration written to .config
#
```

Reply:
2nd approach seems to work.

Reply:
Quoting the Makefile:
```
# Read arch specific Makefile to set KBUILD_DEFCONFIG as needed.
# KBUILD_DEFCONFIG may point out an alternative default configuration
# used for 'make defconfig'
include $(srctree)/arch/$(SRCARCH)/Makefile
export KBUILD_DEFCONFIG KBUILD_KCONFIG CC_VERSION_TEXT

config: outputmakefile scripts_basic FORCE
    $(Q)$(MAKE) $(build)=scripts/kconfig $@

%config: outputmakefile scripts_basic FORCE
    $(Q)$(MAKE) $(build)=scripts/kconfig $@
```

Reply:
I guess the order here is important, so my bad. Doing `make KBUILD_DEFCONFIG=/path defconfig` should work.

Reply:
I think I am just dumb at this point. `make KBUILD_DEFCONFIG=linux.defconfig defconfig`:
```
  HOSTCC  scripts/basic/fixdep
  HOSTCC  scripts/kconfig/conf.o
  HOSTCC  scripts/kconfig/confdata.o
  HOSTCC  scripts/kconfig/expr.o
  LEX     scripts/kconfig/lexer.lex.c
  YACC    scripts/kconfig/parser.tab.[ch]
  HOSTCC  scripts/kconfig/lexer.lex.o
  HOSTCC  scripts/kconfig/menu.o
  HOSTCC  scripts/kconfig/parser.tab.o
  HOSTCC  scripts/kconfig/preprocess.o
  HOSTCC  scripts/kconfig/symbol.o
  HOSTCC  scripts/kconfig/util.o
  HOSTLD  scripts/kconfig/conf
*** Default configuration is based on target 'linux.defconfig'
make[3]: Nothing to be done for 'linux.defconfig'.
```
And no `.config` is produced :(